src/ostree/ot-admin-builtin-finalize-staged.c

```diff
@@ -50,13 +65,57 @@ ot_admin_builtin_finalize_staged (int argc, char **argv, OstreeCommandInvocation
 
   g_autoptr(GOptionContext) context = g_option_context_new ("");
   g_autoptr(OstreeSysroot) sysroot = NULL;
+
+  /* First parse the args without loading the sysroot to see what options are
```

Comment:
I'm OK with this but...I think we could also bypass the whole thing and just do

```
bool opt_hold = argc > 2 && strcmp (argv[1], "--hold") == 0;
```
or so?  Or use an environment variable.

Dunno.  I don't feel really strongly.  My main concern is that someone later comes by and makes some sort of change that isn't ready for `ostree_admin_option_context_parse` being invoked twice.

Reply:
Eh, I'm personally not a fan of adding adhoc option handing. What if the first option is `--verbose`or something else?

I think I can go back to the refactoring route and split out the sysroot opening.

Reply:
I'm OK with the code as is too.

Though...hmm, wouldn't it work to use plain old `g_option_context_parse()` at least?  That would avoid any issues with invoking `ostree_admin_option_context_parse` twice.

Reply:
I think it would except that you'd have to duplicate the handling of the `--sysroot` option at least. I think the refactoring I added to initialize but not load the sysroot should work well. See the new prep commit I added.